### PR TITLE
Started configuring test environment

### PR DESCRIPTION
**Pull request by spiray**
Started configuring test environment based off of the p5,js Sound librarie's config to address #65. 

@MathuraMG @lm-n please review this pr and commit and confirm that this is the path that you want me to go down before i continue. 
1. I installed the necessary modules as devDependencies.
2. I added the tasks to `gruntfile.js`. These are just intital loads and they will need work. 
3. I added the `test` folder.
4. I added some of the env stuff needed to start testing. 

I will start writing tests and push to this pr. Please let me know if you have any feedback or questions.

Thank you!

### CODE REVIEW

**Comment by spiray** [comment]
@MathuraMG @lm-n I have never configured a client side test environment and it seems that i missed a step or 2. Can i schedule a time with one or both of you to get the environment up and running and then i can write tests on my own...?

**Comment by MathuraMG** [comment]
@spiray  - sure, does sometime in the evening this week work for you?

**Comment by spiray** [comment]
Yeah, I can do either tonight after 7:30 or the same time on Thursday.

**Comment by MathuraMG** [comment]
@spiray  I can do 9pm both days!

**Comment by spiray** [comment]
@MathuraMG See the last 2 commits. Can you please merge these if they are all successful for you and then I will continue with the more complex tests within the `gridInterceptor`. These were also just for functionality and i will clean up the code style as i move along.

**Comment by MathuraMG** [comment]
@spiray Sorry I missed this! Doing it today

File: Gruntfile.js [hunk 1]
```diff
@@ -1,41 +1,77 @@
+/* global module */
```
Comment (MathuraMG) [inline Gruntfile.js]: hey @spiray - question - is the tab space set to 4 on your editor or is it the lint settings?

Comment (MathuraMG) [reply Gruntfile.js]: Setting it on eslint is perfect. I think if we make it `"indent": [2, 2]` then, it will be 2 spaces all over.

**Pull request merged**